gst/nnstreamer/nnstreamer_conf.c

```diff
@@ -395,3 +395,77 @@ nnsconf_get_value_bool (nnsconf_type_value type)
 
   return ret;
 }
+
+/**
+ * @brief Internal cache for the custom key-values
+ */
+static GHashTable *custom_table = NULL;
+
+/** @brief Public function defined in the header */
+gchar *
+nnsconf_get_custom_value_string (const gchar * group, const gchar * key)
+{
+  gchar *hashkey = g_strdup_printf ("[%s]%s", group, key);
+  gchar *value = NULL;
+
+  nnsconf_loadconf (FALSE);     /* Load .ini file path */
+
+  if (NULL == custom_table)
+    custom_table = g_hash_table_new (g_str_hash, g_str_equal);
+
+  value = g_hash_table_lookup (custom_table, hashkey);
+
+  if (NULL == value) {
+    gchar *envkey = g_strdup_printf ("NNSTREAMER_%s_%s", group, key);
+
+    /* 1. Read envvar */
+    value = _strdup_getenv (envkey);
+
+    /* 2. Read ini */
+    if (NULL == value && conf.conffile) {
+      g_autoptr (GError) error = NULL;
+      g_autoptr (GKeyFile) key_file = g_key_file_new ();
+
+      if (g_key_file_load_from_file (key_file, conf.conffile, G_KEY_FILE_NONE,
+              &error)) {
+        value = g_key_file_get_string (key_file, group, key, &error);
+      }
+
+      g_key_file_free (key_file);
+
+    }
+
+    if (value)
+      g_hash_table_insert (custom_table, hashkey, value);
+  }
+
+  if (NULL == value)
```

Comment:
In line 414 custom_table = g_hash_table_new (g_str_hash, g_str_equal);
key_destroy_func is null, so passed hashkey will not be freed.

Reply:
All values in custom_table are not destroyed until the end of the process. Anyway, I'll add destroyers.